### PR TITLE
feat: add new CLI tools, make quantity and quantityUnit optional

### LINKED ISSUE #111: bug: order quantity not available in CoA rendered in PDF
# Description

**What keywords did you search in schema-tools issues before filing this one ?**
`PDF`

**Is this a BUG REPORT or FEATURE REQUEST ?**
`BUG`

**What you expected to happen :**
If `Order/Quantity` is not available it and `Order/QuantityUnit` should not be rendered on PDF is it results in `NaN undefined`. This will match the results of the HTML rendering.

Problem if OrderQuantity is unknown

### LINKED ISSUE #114: feat: reduce friction to use schema TS interfaces in PDF generators
# Description

**Is this a BUG REPORT or FEATURE REQUEST ?** FEATURE

**Environment :**

**What you expected to happen :**
Currently to generate TS interfaces out of schema, we download the latest schema version, and call the `generate-interfaces` internal package.

The problem appears when we are updating the PDF rendering before a new release is made. The JSON schema might have changed and so need the TS interface, but the schema is not yet released/published so not available for download.

**Potential solution :**

- [x] Add/create CLI to generate TS interfaces using a local file path.

### PR DESCRIPTION
**Pull request by eamon0989**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Added CLI tools for certificate, HTML and PDF generation.
Updated the `readme` documention the new tools and changes.
Updated `generateContent` script to make `Quantity` and `QuantityUnit` optional.

- Fixes #111 
- Fixes #114 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,3 +63,73 @@ The specification of these libraries can be found in [SEP](https://s1seven.githu
...
+
+Process:
+
+1. Update the schema in question
+2. Check and update the types using the `create-schema-interfaces` CLI tool (paste in the path to the updated schema when prompted)
+3. If necessary, make changes in `generateContent.ts` and run `npm run build`.
+   If `generateContent.ts` has been changed and built, copy the contents of `dist/generateContent.js` to the schema repository and replace the contents of `generate-pdf.min.js` with the new minified code
+4. For a new Release Candidate, add a new fixture (used for testing)
+
+// Insert here instructions on how to create a new fixture
```
Comment (getlarge) [inline README.md:85]: no need for this comment anymore

File: README.md [hunk 2]
```diff
@@ -36,3 +63,73 @@ The specification of these libraries can be found in [SEP](https://s1seven.githu
...
+### Fixtures
+
+- Create or use a folder following the schema name
+- Create a subfolder following the version of the release candidate
+- Add a valid json certificate with the name `valid_cert.json`
+- Add an invalid json certificate with the name `invalid_cert.json`
+- Add the `translations.json` - for ease of use we keep English and German translations in a single file
+- `certificate.ts`, `template_hbs.html` and `valid_cert.pdf` are dynamically generated with the following scripts:
+
+To generate `certificate.ts`, run
```
Comment (getlarge) [inline README.md:109]: you could add info about what this file is, like To generate `certificate.ts` (json-schema converted to Typescrict interfaces)

File: README.md [hunk 3]
```diff
@@ -36,3 +63,73 @@ The specification of these libraries can be found in [SEP](https://s1seven.githu
...
+
+```sh
+npm run fixtures:interfaces -- -s /Users/eamon/work/CoA-schemas/schema.json -o fixtures/CoA/v0.1.1/certificate.ts
+```
+
+from the root directory where -s indicates the path to the updated schema and -o indicates the path to the certificate to be generated.
+
+To generate `template_hbs.html`, run
```
Comment (getlarge) [inline README.md:117]: Same as above like To generate `template_hbs.html` (HTML certificate generated from Handlebars template)

File: README.md [hunk 4]
```diff
@@ -36,3 +63,73 @@ The specification of these libraries can be found in [SEP](https://s1seven.githu
...
+
+```sh
+npm run fixtures:html -- -c fixtures/CoA/v0.1.1/valid_cert.json -o fixtures/CoA/v0.1.1/template_hbs.html -t fixtures/CoA/v0.1.1/translations.json -T /Users/eamon/work/CoA-schemas/template.hbs
+```
+
+from the root directory.
+
+To generate `valid_cert.pdf`, run
```
Comment (getlarge) [inline README.md:125]: Same as above like To generate `valid_cert.pdf` (PDF certificate generated from JS generator)

File: utils/create-pdf-certificate.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+})();
+
+// export declare function generatePdf(certificateInput: Record<string, unknown> | string, options: {
+//   outputType: 'stream';
+//   inputType?: 'json' | 'html';
+//   generatorPath?: string;
+//   docDefinition?: Partial<TDocumentDefinitions>;
+//   fonts?: TFontDictionary;
+//   translations?: Translations;
+// }): Promise<PDFKit.PDFDocument>;
```
Comment (getlarge) [inline utils/create-pdf-certificate.js:154]: Please delete this comment

File: utils/create-html-certificate.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  const argv = yargs(hideBin(process.argv))
+    .usage('Usage: $0 -s [schemaPath] -o [outputPath] -t [translationsPath] -T [templatePath]')
+    .options({
+      certificatePath: {
+        description: 'Path to the schema',
+        demandOption: true,
+        type: 'string',
+        coerce: (certificatePath) => {
+          if (!fs.existsSync(path.resolve(certificatePath))) {
+            throw new Error('Certificate file path does not exist.');
```
Comment (getlarge) [inline utils/create-html-certificate.js:23]: I don't think we need this file anymore right ?

**Review by getlarge — CHANGES_REQUESTED** [review]

File: utils/create-html-certificate.js (hunk 6 above)
Comment (eamon0989) [reply utils/create-html-certificate.js:23]: This is one of the tools we made yesterday using yargs, are you sure we don't need it?

Comment (getlarge) [reply utils/create-html-certificate.js:23]: My mistake, the culprit seems to be [packages/generate-coa-pdf-template/utils/create-schema-interfaces-interactive.js](https://github.com/s1seven/schema-tools/pull/116/files#diff-773746113e99e4b7615605e4bbb8b2636fa9d8ae297543cd93da9c26eef2cdf9)

Comment (eamon0989) [reply utils/create-html-certificate.js:23]: No problem, that's done 👍🏼

File: packages/generate-coa-pdf-template/package.json [hunk 7]
```diff
@@ -44,8 +44,7 @@
     "test:staged": "jest --runInBand --onlyChanged",
     "scan:deps": "audit-ci --config ../../audit-ci.json",
     "create-example": "node ./utils/create-example.js",
-    "create-interfaces": "node ./utils/create-schema-interfaces.js https://schemas.s1seven.com/coa-schemas/v0.1.0/schema.json ./src/types/schemaTypes.ts",
-    "create-interfaces-interactive": "node ../../utils/create-schema-interfaces-interactive.js ./src/types/schemaTypes.ts"
```
Comment (getlarge) [inline packages/generate-coa-pdf-template/package.json]: You did not need to remove `create-interfaces-interactive`, as you were using the `utils` located in the root

Comment (eamon0989) [reply packages/generate-coa-pdf-template/package.json]: My bad, I thought it was outdated, but looking back at it I see I was wrong. I've added it back in.

**Review by getlarge — APPROVED** [review]

**Pull request merged**